app/validators/attachment_checksum_validator.rb

```diff
@@ -2,18 +2,28 @@
 
 # Validator for Attachment checksum
 class AttachmentChecksumValidator < ActiveModel::Validator
-  def validate(record)
+  def validate(record) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
     return if record.file.checksum.blank?
 
     klass = record.class
 
     # add error if another Attachment associated with the Attachable exists with the same checksum
+
+    # checks if file has same checksum and name (no error if same contents but different name)
     if klass.joins(:file_blob)
             .where.not(id: record.id)
             .exists?(attachable_id: record.attachable_id,
                      attachable_type: record.attachable_type,
                      deleted_at: nil,
-                     file_blob: { checksum: record.file.checksum })
+                     file_blob: { checksum: record.file.checksum, filename: record.file.filename.to_s }) ||
```

Comment:
**nitpick (blocking):** No need to have a `||` since we only want to add the error when an existing attachment exists with the same filename and checksum. If a file already exists with the same name but different checksum we can allow attachment as that is existing behaviour.

Reply:
Fixed in 163e8b5fe23ffeb5377469c8f6b1f57dddf182a8